core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -436,6 +444,14 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         final RoutingResult routingResult = routed.routingResult();
         final ServiceConfig serviceCfg = routed.value();
         final HttpService service = serviceCfg.service();
+        // Respond BAD_REQUEST if the proxied addresses are invalid.
+        if (invalidProxiedAddressesException != null) {
+            responseEncoder.writeErrorResponse(req.id(), req.streamId(), serviceCfg, headers,
+                    HttpStatus.BAD_REQUEST, "Invalid proxied address",
+                    invalidProxiedAddressesException);
+            decreasePendingRequests();
```

Comment:
I do think `req.abort()` should be called from somewhere - otherwise, the remote stream may continue to be opened unintentionally.